internal/mutators/podtemplatespec_test.go

```diff
@@ -0,0 +1,407 @@
+			wantLimits: corev1.ResourceList{},
+		},
+		{
+			msg:        "Memory limit already exists, do not set",
+			requests:   corev1.ResourceList{},
+			limits:     corev1.ResourceList{corev1.ResourceMemory: resource.MustParse("10Mi")},
+			mc:         memoryConfigWithMaxRatio,
+			wantLimits: corev1.ResourceList{corev1.ResourceMemory: resource.MustParse("10Mi")},
+		},
+		{
+			msg:        "No maxLimitRequestMemoryRatio set, use defaultMemoryLimit which is higher than defaultLimitRequestMemoryRatio",
+			requests:   corev1.ResourceList{corev1.ResourceMemory: resource.MustParse("10Mi")},
+			limits:     corev1.ResourceList{},
+			mc:         memoryConfigWithoutMaxRatio,
+			wantLimits: corev1.ResourceList{corev1.ResourceMemory: *quantity.Ptr(resource.MustParse("50Mi")).ToDec()},
+		},
+		{
+			msg:        "No maxLimitRequestMemoryRatio set, use defaultLimitRequestMemoryRatio which is higher than defaultMemoryLimit",
+			requests:   corev1.ResourceList{corev1.ResourceMemory: resource.MustParse("49Mi")},
+			limits:     corev1.ResourceList{},
+			mc:         memoryConfigWithoutMaxRatio,
+			wantLimits: corev1.ResourceList{corev1.ResourceMemory: resource.MustParse("53.9Mi")},
+		},
+		{
+			msg:        "MaxLimitRequestMemoryRatio set, defaultMemoryLimit and defaultLimitRequestMemoryRatio both larger, use MaxLimitRequestMemoryRatio",
+			requests:   corev1.ResourceList{corev1.ResourceMemory: resource.MustParse("45Mi")},
+			limits:     corev1.ResourceList{},
+			mc:         memoryConfigWithMaxRatio,
+			wantLimits: corev1.ResourceList{corev1.ResourceMemory: resource.MustParse("47.25Mi")},
+		},
+		{
+			msg:      "MaxLimitRequestMemoryRatio set, defaultMemoryLimit and defaultLimitRequestMemoryRatio both smaller, use MaxLimitRequestMemoryRatio",
+			requests: corev1.ResourceList{corev1.ResourceMemory: resource.MustParse("50Gi")},
+			limits:   corev1.ResourceList{},
+			mc: limitrange.Config{
+				HasDefaultLimit:         true,
+				HasMaxLimitRequestRatio: true,
+				DefaultLimit:            resource.MustParse("50Mi"),
+				MaxLimitRequestRatio:    resource.MustParse("1.3"),
+			},
+			wantLimits: corev1.ResourceList{corev1.ResourceMemory: *quantity.Ptr(resource.MustParse("65Gi")).ToDec()},
+		},
+	}
+
+	for _, test := range tests {
+		container := &corev1.Container{
+			Resources: corev1.ResourceRequirements{
+				Requests: test.requests,
+				Limits:   test.limits,
+			},
+		}
+
+		wantContainer := &corev1.Container{
+			Resources: corev1.ResourceRequirements{
+				Requests: test.requests,
+				Limits:   test.wantLimits,
+			},
+		}
+
+		err := testPts.setMemoryLimit(container, test.mc)
+		assert.NoError(t, err, test.msg)
+
+		// round up to scale 0 to avoid Scale differences
+		roundUpContainerQuantityToScale(container, corev1.ResourceMemory, 0)
+		roundUpContainerQuantityToScale(wantContainer, corev1.ResourceMemory, 0)
+
+		assert.Equal(t, wantContainer, container, test.msg)
+	}
+}
+
+// convenience function for testing only
```

Comment:
nit: These comments are unnecessary given these functions are in a testing file.